src/util/bip158.rs

```diff
@@ -144,18 +145,20 @@ impl BlockFilter {
     }
 
     /// Returns true if any query matches against this [`BlockFilter`].
-    pub fn match_any<'a, I>(&self, block_hash: &BlockHash, query: I) -> Result<bool, Error>
+    pub fn match_any<I>(&self, block_hash: &BlockHash, query: I) -> Result<bool, Error>
     where
-        I: Iterator<Item = &'a [u8]>,
+        I: Iterator,
+        I::Item: AsRef<[u8]>,
```

Comment:
`Borrow` feels more correct but I'm not sure if there's any problem with `AsRef`

Reply:
I'm not fully groking the why to favour one over the other but I'm going to defer to you on this one. Changed to `Borrow`.

Reply:
Just paranoia, really. I was too tired when I wrote it and `AsRef` probably would've been fine.